### ISSUE TITLE
Cannot get reproducible build to match

### ISSUE THREAD

**Issue by dancek** [signalapp/Signal-Android#13754]
### Guidelines

- [X] I have searched searched open and closed issues for duplicates
- [X] I am submitting a bug report for existing functionality that does not work as intended
- [X] This isn't a feature request or a discussion topic

### Bug description

I tried checking that my installed android app (from Play Store) matches a local reproducible build. It didn't seem to match.

I might be overlooking something obvious, please bear with me if that is the case. I'm not very experienced with apks.

The app on the device reports version 7.20.2. The pulled base apk and git repo are:
```
❯ sha256sum play/base.apk
d17bc0f622d72a1ad65388814e50eaa2167288bc5cb6f4bd3f1f9110e32d44cc  play/base.apk
❯ git describe
v7.20.2
```

The build worked fine. When I compared the result with apkdiff, I get

```
❯ python apkdiff/apkdiff.py me/base-master.apk play/base.apk
Manifest lengths differ!
Sorted manifests don't match, META-INF/Signal-Android_playProdRelease.kotlin_module vs META-INF/BNDLTOOL.RSA
APKs don't match!
```

Adding  BNDLTOOL.RSA and BNDLTOOL.SF to ignores, I get
```
❯ python apkdiff/apkdiff.py me/base-master.apk play/base.apk
Manifest lengths differ!
Sorted manifests don't match, META-INF/collection.kotlin_module vs META-INF/code_transparency_signed.jwt
APKs don't match!
```

Skipping compareEntryNames altogether and commenting out all `return False`s gets me
```
❯ python apkdiff/apkdiff.py me/base-master.apk play/base.apk
APK info lists of different length!
APKs differ on file AndroidManifest.xml! Files extracted to the mismatches/ directory.
APKs differ on file assets/dexopt/baseline.prof! Files extracted to the mismatches/ directory.
APKs differ on file assets/dexopt/baseline.profm! Files extracted to the mismatches/ directory.
APKs differ on file classes.dex! Files extracted to the mismatches/ directory.
APKs differ on file classes2.dex! Files extracted to the mismatches/ directory.
APKs differ on file classes3.dex! Files extracted to the mismatches/ directory.
APKs differ on file classes4.dex! Files extracted to the mismatches/ directory.
APKs differ on file classes5.dex! Files extracted to the mismatches/ directory.
APKs differ on file res/xml/splits0.xml! Files extracted to the mismatches/ directory.
APKs differ on file resources.arsc! Files extracted to the mismatches/ directory.
APKs don't match!
```

---

Relevant software versions I used for this:
```
❯ git --version
git version 2.47.0
❯ docker --version
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
podman version 5.2.5
❯ python --version
Python 3.12.7
❯ adb --version
Android Debug Bridge version 1.0.41
Version 34.0.4-10411341
Installed as /opt/android-sdk/platform-tools/adb
Running on Linux 6.11.5-arch1-1 (x86_64)
❯ bundletool version
1.17.2
```

I'd expect this to be a version issue. Probably someone can check the apks don't differ for them, look at software versions and make an educated guess as to what breaks the reproducibility here. But I'm reporting it so the requirements can be documented better.

### Device

Oneplus 11 5G

### Android version

14

### Signal version

7.20.2

**Comment by signal-support-request** [comment]
Possibly related: https://github.com/mollyim/mollyim-android/issues/338

**Comment by greyson-signal** [comment]
Hello! Thank you for reporting this. We dug into it and found some non-deterministic proguarding behavior resulting from  some navigation graph stuff. This has been addressed and should be working again in 7.23.

(Oh also, we forgot to exclude some of the new bundle-related signing files, which will also be addressed in 7.23)

Thank you!
